src/models/ghost.c

```diff
@@ -144,4 +144,103 @@ bool isGhostEatableRunningOut()
 int getEatenGhostScore(int ghostEaten)
 {
     return pow(2, ghostEaten) * 100;
+}
+
+void moveGhost(struct Ghost *sprite)
+{
+    updatePosition(&sprite->uiPosition, sprite->direction, SPRITE_SPEED);
+
+    struct Position updatedGridPos = uiPosToGridPos(getCellCenter(sprite->uiPosition));
+
+    if (arePositionEquals(sprite->gridPosition, updatedGridPos))
+        return;
+
+    sprite->gridPosition = updatedGridPos;
+    onGhostGridPositionChanged(sprite);
+
+}
+
+void onGhostGridPositionChanged(struct Ghost *sprite)
+{
+    // Todo : change direction when possible without collision
+    //sprite->direction = sprite->nextDirection;
+    selectNextGhostDirection(sprite);
+
+    // TODO : ghost will turn instantly but will go through walls. We should implement ghost whished direction + obstacle detection like for pacman
+}
+
+void setGhostTargetTile(struct Ghost *sprite, struct Position targetTile)
+{
+    sprite->targetTile = targetTile;
+}
+
+void selectNextGhostDirection(struct Ghost *sprite)
+{
+    struct Position nextPosition = sprite->gridPosition;
+    updatePosition(&nextPosition, sprite->direction, 1);
+
+    // Get possible directions
+    Direction possibleDirections[DIRECTION_COUNT-1];
+    int possibleDirectionsCount = 0;
+
+    for (int i = 0; i < DIRECTION_COUNT; i++)
+    {
+        Direction direction = i;
+
+        // if is opposite direction don't att
+        if (direction == getOppositeDirection(sprite->direction))
+            continue;
+
+        struct Position cell = sprite->gridPosition;
+        updatePosition(&cell, direction, 1);
+        // if is obstacle don't add
+        if (isObstacle(cell))
+            continue;
+
+        // add direction if previous conditions passed
+        possibleDirections[possibleDirectionsCount] = direction;
+        possibleDirectionsCount++;
+    }
+
+    // if 1 direction possible, select it
+    if (possibleDirectionsCount == 1)
+    {
+        sprite->nextDirection = possibleDirections[0];
+        return;
+    }
+
+    // else if 2 directions possible, select the one that is the closest to the target tile
+    int minimumDistance = 1000000;
```

Comment:
why this number, can't we create a const ?

Reply:
it's just a number bigger than any possible distance
maybe ...

Reply:
so that's a magic number ?